config/routes.rb

```diff
@@ -1,8 +1,19 @@
 Rails.application.routes.draw do
+  root to: 'companies#index'
+
+  resources :categories
 
   resources :companies do
-    resources :jobs
+    resources :contacts
+    resources :jobs do
+      resources :comments, only: [:create, :new]
```

Comment:
i suggest only nesting two deep when nesting your routes. they can get long and a little confusing :(